File: 지연(이)/24년 2월 19일/PGS 기능개발.cpp
#include <string>
#include <vector>
#include<iostream>
#include<cmath>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int N = progresses.size();
    
    vector<int> stack;
    for(int i=0;i<N;i++){
        int tmp = ((100 - progresses[i]) / speeds[i]);
        if ((100-progresses[i])%speeds[i]!=0){
            tmp++;
        }
        stack.push_back(tmp);
    }
    
    int cnt = 1;
    int mx = stack[0];
    for(int i=1;i<N;i++){
        if(mx<stack[i]){
            answer.push_back(cnt);
            cnt=1;
            mx = stack[i];
        }
        else{
            cnt++;
        }
    }
    answer.push_back(cnt);
    
    return answer;
}

File: 지연(이)/24년 2월 19일/PGS 기능 개발.py
import math

def solution(progresses, speeds):
    answer = []
    N = len(progresses)
    
    stack = []
    for i in range(N):
        stack.append(math.ceil((100-progresses[i])/speeds[i]))
    
    cnt = 1
    mx = stack[0]
    for i in range(1,N):
        if mx<stack[i]:
            answer.append(cnt)
            cnt=1
            mx = stack[i]
        else:
            cnt+=1
    answer.append(cnt)
    
    return answer
